crates/vm2/src/instruction_handlers/far_call.rs

```diff
@@ -182,15 +185,15 @@ pub(crate) fn get_far_call_arguments(abi: U256) -> FarCallABI {
 ///
 /// This function needs to be called even if we already know we will panic because
 /// overflowing start + length makes the heap resize even when already panicking.
-pub(crate) fn get_far_call_calldata<T: Tracer, W>(
+pub(crate) fn get_far_call_calldata<T: Tracer, W: World<T>>(
     raw_abi: U256,
     is_pointer: bool,
     vm: &mut VirtualMachine<T, W>,
     already_failed: bool,
 ) -> Option<FatPointer> {
     let mut pointer = FatPointer::from(raw_abi);
 
-    match FatPointerSource::from_abi((raw_abi.0[3] >> 32) as u8) {
+    match FatPointerSource::from_abi((raw_abi.0[3] >> 32) & 0xff) {
```

Comment:
I prefer the version that clippy dislikes here. Maybe that lint should be turned off.